src/ripple/protocol/KnownFormats.h

```diff
@@ -35,7 +36,7 @@ namespace ripple {
 
     @tparam KeyType The type of key identifying the format.
 */
-template <class KeyType>
+template <class KeyType, class Derived>
```

Comment:
The only reason to use the CRTP is to initialize the `name_` data member.  CRTP seems like overkill for such a use case. I'd prefer that we just pass the name to the constructor. This will always be done from the constructor from the derived class, so it's still easy for callers to use.
As a side note, if you did want to use the CRTP, you don't need to store the member.